wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -124,81 +134,330 @@
     "components": {
       "type": "array",
       "title": "The Components List",
-      "description" : "Description of the workspace components, such as editor and plugins",
+      "description": "Description of the workspace components, such as editor and plugins",
       "items": {
         "type": "object",
         "required": [
           "type"
         ],
-        "additionalProperties": false,
-        "oneOf" : [
-          {
-            "properties": {
-              "type": {
-                "enum": [
-                  "cheEditor",
-                  "chePlugin"
-                ]
-              },
-              "alias": {},
-              "id": {},
-              "memoryLimit": {}
-            },
-            "required": [
-              "id"
-            ],
-            "additionalProperties": false
+        "if": {
+          "properties": {
+            "type": {
+              "type": "string"
+            }
           },
-          {
-            "properties": {
-              "type": {
-                "enum": [
-                  "kubernetes",
-                  "openshift"
-                ]
+          "required": ["type"]
+        },
+        "then": {
+          "allOf": [
+            {
+              "if": {
+                "properties": {
+                  "type": {
+                    "enum": [
+                      "cheEditor",
+                      "chePlugin"
+                    ]
+                  }
+                }
               },
-              "alias": {},
-              "reference": {},
-              "referenceContent": {},
-              "selector": {},
-              "entrypoints": {}
+              "then": {
+                "required": [
+                  "id"
+                ],
+                "additionalProperties": false,
+                "properties": {
+                  "type": {},
+                  "alias": {},
+                  "id": {
+                    "type": "string",
+                    "description": "Describes the component id. It has the following format: [{REGISTRY_URL}/]{plugin/editor ID}:{plugin/editor VERSION}, where `{REGISTRY_URL}/` is an optional part.",
```

Comment:
Looks like I forgot to change the description of this field in my PR about naming convention changes. Could you fix it in your PR?
It should be something like:
`Describes the component id. It has the following format: [{REGISTRY_URL}/]{plugin/editor PUBLISHER}/{plugin/editor NAME}/{plugin/editor VERSION}, where `{REGISTRY_URL}/` is an optional part.`

Reply:
And, please, fix examples too

Reply:
ok